src/Streamly/Internal/Data/Array/Foreign.hs

```diff
@@ -459,8 +459,9 @@ getIndex arr i =
 --
 -- /Unimplemented/
 {-# INLINE getIndices #-}
-getIndices :: Unfold m (Array a) Int -> Unfold m (Array a) a
-getIndices = undefined
+getIndices :: (MonadIO m, Storable a)
+    => SerialT m Int -> Unfold m (MA.Array a) a
+getIndices (SerialT stream) = MA.getIndices $ D.fromStreamK stream
```

Comment:
This is immutable array module, this API should use the immutable array type, not `MA.Array`.